gcs/gcscaching/fast_stat_bucket.go

```diff
@@ -187,8 +187,10 @@ func (b *fastStatBucket) ComposeObjects(
 func (b *fastStatBucket) StatObject(
 	ctx context.Context,
 	req *gcs.StatObjectRequest) (o *gcs.Object, err error) {
-	// Do we have an entry in the cache?
-	if hit, entry := b.lookUp(req.Name); hit {
+	// Check if we need to return from cache. If yes, Do we have an entry
+	// in the cache?
+	hit, entry := b.lookUp(req.Name)
```

Comment:
Why even look into the cache when force fetch is enabled?

Reply:
Since its a map lookup, added the condition to same if clause.
 Refactored the code now.